src/custom/state/claim/hooks/index.ts

```diff
@@ -94,3 +113,178 @@ export function useUserUnclaimedAmount(account: string | null | undefined): Curr
+      totalValue = JSBI.add(totalValue, JSBI.BigInt(value))
+    }
+  })
+
+  return {
+    args: [indices, claimTypes, claimants, claimableAmounts, claimedAmounts, merkleProofs, sendEth],
+    totalClaimedAmount,
+    value: totalValue.toString() === '0' ? undefined : totalValue.toString(),
+  }
+}
+
+/**
+ * Hook that returns the claimCallback
+ *
+ * Different from the original version, the returned callback takes as input a list of ClaimInputs,
+ * which is an object of the claim index and the amount being claimed.
+ *
+ * @param account
+ */
+export function useClaimCallback(account: string | null | undefined): {
+  claimCallback: (claimInputs: ClaimInput[]) => Promise<string | undefined>
+} {
+  // get claim data for this account
+  const { chainId, account: connectedAccount } = useActiveWeb3React()
+  const claims = useUserAvailableClaims(account)
+  const vCowContract = useVCowContract()
+
+  // used for popup summary
+  const addTransaction = useTransactionAdder()
+  const vCowToken = chainId ? V_COW[chainId] : undefined
+
+  const claimCallback = useCallback(
+    async function (claimInput: ClaimInput[]) {
+      if (
+        claims.length === 0 ||
+        claimInput.length === 0 ||
+        !account ||
+        !connectedAccount ||
+        !chainId ||
+        !vCowContract ||
+        !vCowToken
+      ) {
+        console.log(
+          `claimHooks::claimCallback::something is missing`,
+          claims.length === 0,
+          claimInput.length === 0,
+          !account,
+          !connectedAccount,
+          !chainId,
+          !vCowContract,
+          !vCowToken
+        )
+        throw new Error("Not initialized, can't claim")
+      }
+      console.log(`claimHooks::claimCallback::got through`, claims, claimInput, account, connectedAccount, chainId)
+
+      const { args, totalClaimedAmount, value } = _buildArgs({ claimInput, claims, account, connectedAccount })
+
+      console.log(`claimHooks::claimCallback::args`, JSON.stringify(args, undefined, 4))
+      // const argsSingle: [number, ClaimType, string, BigNumber, BigNumber, string[]] = []
+
+      if (args[0].length === 0) {
+        console.log(`claimHooks::claimCallback::no args`)
+        throw new Error('There are no valid claims')
+      }
+
+      const vCowAmount = CurrencyAmount.fromRawAmount(vCowToken, totalClaimedAmount)
+
+      // return vCowContract
+      //   .claim(args[0][0], args[1][0], args[2][0], args[3][0], args[4][0], args[5][0], {
+      //     value,
+      //     from: connectedAccount,
+      //   })
+      //   .then((response: TransactionResponse) => {
+      //     addTransaction({
+      //       hash: response.hash,
+      //       summary: `Claimed ${formatSmart(vCowAmount)} vCOW`,
+      //       claim: { recipient: account },
+      //     })
+      //     return response.hash
+      //   })
+
+      return vCowContract.estimateGas['claimMany'](...args, { value }).then((estimatedGas) => {
```

Comment:
This call fails with the error:

Error: cannot estimate gas; transaction may fail or may require manual gas limit (error={"code":-32603,"message":"execution reverted","data":{"originalError":{"code":3,"data":"0x09bde339","message":"execution reverted"}}}, method="estimateGas", transaction={"from":"0x5B0ABE214aB7875562ADeE331dEfF0Fe1912FE42","to":"0x8F07778b703F9FEB93521FC5230e859Dbd032b6b","data":"0xf3507dab00000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000120000000000000000000000000000000000000000000000000000000000000016000000000000000000000000000000000000000000000000000000000000001a000000000000000000000000000000000000000000000000000000000000001e000000000000000000000000000000000000000000000000000000000000002200000000000000000000000000000000000000000000000000000000000000460000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000004380000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000ae21f9700ed15e1a83baa3183a67d54b1da208600000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000001f78ec55cc4cb84000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000001f78ec55cc4cb84000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000f03cc3df79d865ad3bfbdd54ec8ee6001f294655fe0018e660c0879c046d82133d044fdd7448e035e36a13a1b3edb2d7158624c6578a31517e8deefc2034c216b4531947895cb7cfc7d878e32288a73c04eaf63af8345d47283ba2cb8bac7e8c5f1376d3d11b33948440cf71bcfa63e69a9ca436ab3cef016b9da36f2a226d3e2a5ff51dde0385f442354e8d2582e81fd7d81f0a9b558ceea8e84136e6b88501cdc13f692869b72e45579e0ef7f5afe670ffeb5c2931f5b72cc50a19121c57f68bf454c5b68f2a3637a1221ce1e5834ce5725f9fcef7f932f81af24127e312d580c85968537348d6281fb14be6b4528e77fce7adb1ac9a6fae52c83182f64d848dc838baad0c5b2844072e96cd5376ca337f80b42746b46a46121afcc0e39e0ee52aaac536243500f3d5aae091887032322773b40f7fa906d0043a5d74013a8c457158a64c297f7a0ae2530774437f91667aab006e084f3214986dce3ca898474c1d3779a02affbc6b4885b248a636f9a3fd124d356e639347bb04129436133bde0b4fbf77c444bada1498ecbee1431eb297d4fa9a6d04f1dcb19571c334b5dcb768e2f52bd7f4c8f1251620800bed30d14d13453dd2f0ced98eff0212d5e52883fb9f3c87451effbb58340761f30e2d2057a0a2fb27eee5ec6af3362be0389f900000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000","accessList":null}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.4.5)